lib/Horde/Core/VersionChecker.php

```diff
@@ -0,0 +1,485 @@
+        $composerFile = $this->rootDir . '/composer.json';
+        
+        if (!file_exists($composerFile)) {
+            throw new Horde_Exception('Could not find composer.json in ' . $this->rootDir);
+        }
+
+        // Load installed.json
+        $installedPath = $this->rootDir . '/vendor/composer/installed.json';
+        if (!file_exists($installedPath)) {
+            throw new Horde_Exception('Could not find installed.json at ' . $installedPath);
+        }
+
+        $installedData = json_decode(file_get_contents($installedPath), true);
+        if ($installedData === null) {
+            throw new Horde_Exception('Could not parse installed.json at ' . $installedPath);
+        }
+
+        $this->installedPackages = $installedData['packages'] ?? $installedData;
+        if (!is_array($this->installedPackages)) {
+            throw new Horde_Exception('Invalid format in installed.json at ' . $installedPath);
+        }
+
+       //Horde::Log('\Composer\InstalledVersions::getAllRawData(): ' . print_r(\Composer\InstalledVersions::getAllRawData(), true), 'ERR');
+        
+        try {
+            // Create Composer instance with proper working directory
+            $this->io = new \Composer\IO\NullIO();
+            $this->config = \Composer\Factory::createConfig($this->io, $this->rootDir);
+            $this->composer = \Composer\Factory::create($this->io, $composerFile, false, $this->rootDir, $this->config);
+            $this->repositoryManager = $this->composer->getRepositoryManager();
+            $this->process = new \Composer\Util\ProcessExecutor($this->io);
+            $this->httpDownloader = new \Composer\Util\HttpDownloader($this->io, $this->config);
+            $this->eventDispatcher = new \Composer\EventDispatcher\EventDispatcher($this->composer, $this->io);
+        } catch (\Exception $e) {
+            throw new Horde_Exception('Failed to initialize Composer: ' . $e->getMessage());
+        }
+    }
+
+    /**
+     * Get the installed 'horde/*' packages. other packages are filtered out.
+     *
+     * This method is static to avoid creating a new instance of the VersionChecker class by performance reasons.
+     * 
+     * @return array[] Array with package names (without horde/ prefix) as keys and package info as values
+     */
+    public static function getInstalledPackages(): array
+    {
+        $packages = array();
+        try {
+            // Get all installed packages
+            foreach (\Composer\InstalledVersions::getInstalledPackages() as $packageName) {
+
+                Horde::Log('Installed package: ' . print_r($packageName, true), 'ERR');
+                // Only check Horde packages
+                if (str_starts_with($packageName, 'horde/')) {
+                    $shortName = substr($packageName, 6); // Remove 'horde/' prefix
+                    $packages[$shortName] = array(
+                        'version' => \Composer\InstalledVersions::getVersion($packageName),
```

Comment:
Consider using the registry instead

Reply:
Isn't handling the registry just and only the applications itself but not all other horde/* libraries?
My initial thought to handle the version /update management via the installation vector instead of via the application itself, but I'm open for any other approach :)